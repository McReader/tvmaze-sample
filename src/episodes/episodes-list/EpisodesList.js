import GridList from '@material-ui/core/GridList';

import { RequestStatus } from "../../ui-lib/request-status/RequestStatus";

import { EpisodeListItem } from "./item/EpisodeListItem";

import "./EpisodesList.css";

export function EpisodesList({ currentUrl = "", episodes = [], requestStatus }) {
    return (
        <div className="EpisodesList">
            <RequestStatus requestStatus={requestStatus}>
                {() => (
                    <GridList className="EpisodesListGrid" spacing={1}>
                        {episodes.map(episode => (
                            <EpisodeListItem
                                currentUrl={currentUrl}
                                episode={episode}
                                key={episode.id}
                            />
                        ))}
                    </GridList>
                )}
            </RequestStatus>
        </div>
    );
}
