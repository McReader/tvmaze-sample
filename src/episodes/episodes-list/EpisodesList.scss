.EpisodesList {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-around;
	overflow: hidden;
}

.EpisodesListGrid {
	flex-wrap: nowrap;
	/* Promote the list into his own layer on Chrome. This cost memory but helps keeping high FPS. */
	transform: translateZ(0);
}
