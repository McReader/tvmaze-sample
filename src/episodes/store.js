import { createEntityAdapter, createAsyncThunk, createSlice } from "@reduxjs/toolkit";

const PREFIX = "episodes";

export const fetchShowEpisodeList = createAsyncThunk(`${PREFIX}/fetchShowEpisodeList`, async (showId, thunkAPI) => {
    const result = await thunkAPI.extra.showsService.getShowEpisodeList(showId);
    return result.data;
});

export const fetchEpisodeMainInformation = createAsyncThunk(`${PREFIX}/fetchEpisodeMainInformation`, async ({ showId, episodeId }, thunkAPI) => {
    const episode = episodesSelectors.selectById(thunkAPI.getState(), episodeId);

    if (episode) {
        return Promise.resolve(episode);
    }

    const result = await thunkAPI.extra.episodesService.getEpisodeMainInformation(episodeId);
    return result.data;
});

const episodesAdapter = createEntityAdapter();

export const initialState = {
    ...episodesAdapter.getInitialState(),
    byShow: {},
};

export const episodesSlice = createSlice({
    name: PREFIX,
    initialState: initialState,
    reducers: {
        // standard reducer logic, with auto-generated action types per reducer
    },
    extraReducers: {
        [fetchShowEpisodeList.fulfilled]: (state, { meta, payload }) => {
            episodesAdapter.addMany(state, payload);
            state.byShow[meta.arg] = payload.map(episode => episode.id);
        },
        [fetchEpisodeMainInformation.fulfilled]: (state, { meta, payload }) => {
            episodesAdapter.addOne(state, payload);
            state.byShow[meta.arg.showId] = [payload.id];
        },
    }
});

export const episodesSelectors = {
    ...episodesAdapter.getSelectors((state) => state.episodes),
    selectByShowId: (state) => {  },
};
