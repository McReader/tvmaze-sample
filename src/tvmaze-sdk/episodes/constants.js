export const EPISODES_RESOURCE_PATH = "/episodes";
