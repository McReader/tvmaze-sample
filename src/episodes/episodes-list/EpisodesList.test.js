import renderer from "react-test-renderer"
import { MemoryRouter } from "react-router-dom"

import { EpisodesList } from "./EpisodesList"

import episodes from "../../tvmaze-sdk/episodes/episodesSample.json"

import { REQUEST_STATUS_FULFILLED } from "../../ui-lib/request-status/RequestStatus"

let tree
let props

beforeEach(() => {
	tree = renderer
		.create(
			<MemoryRouter>
				<EpisodesList {...props} />
			</MemoryRouter>
		)
		.toJSON()
})

describe(`when request status is "${REQUEST_STATUS_FULFILLED}" and there are episodes in list`, () => {
	beforeAll(() => {
		props = {
			requestStatus: REQUEST_STATUS_FULFILLED,
			episodes,
		}
	})

	test("should match snapshot", () => {
		expect(tree).toMatchSnapshot()
	})
})
