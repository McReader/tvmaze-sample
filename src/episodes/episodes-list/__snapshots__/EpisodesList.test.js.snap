// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when request status is "fulfilled" and there are episodes in list should match snapshot 1`] = `
<div
  className="EpisodesList"
>
  <ul
    className="MuiGridList-root EpisodesListGrid"
    style={
      Object {
        "margin": -0.5,
      }
    }
  >
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/1"
          onClick={[Function]}
        >
          <img
            alt="Pilot"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4388.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Pilot
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E1
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/2"
          onClick={[Function]}
        >
          <img
            alt="The Fire"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4389.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                The Fire
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E2
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/3"
          onClick={[Function]}
        >
          <img
            alt="Manhunt"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4390.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Manhunt
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E3
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/4"
          onClick={[Function]}
        >
          <img
            alt="Outbreak"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4391.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Outbreak
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E4
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/5"
          onClick={[Function]}
        >
          <img
            alt="Blue on Blue"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4392.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Blue on Blue
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E5
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/6"
          onClick={[Function]}
        >
          <img
            alt="The Endless Thirst"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4393.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                The Endless Thirst
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E6
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/7"
          onClick={[Function]}
        >
          <img
            alt="Imperfect Circles"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4394.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Imperfect Circles
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E7
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/8"
          onClick={[Function]}
        >
          <img
            alt="Thicker Than Water"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4395.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Thicker Than Water
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E8
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/9"
          onClick={[Function]}
        >
          <img
            alt="The Fourth Hand"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4396.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                The Fourth Hand
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E9
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/10"
          onClick={[Function]}
        >
          <img
            alt="Let the Games Begin"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4397.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Let the Games Begin
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E10
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/11"
          onClick={[Function]}
        >
          <img
            alt="Speak of the Devil"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4398.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Speak of the Devil
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E11
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/12"
          onClick={[Function]}
        >
          <img
            alt="Exigent Circumstances"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4399.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Exigent Circumstances
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E12
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/13"
          onClick={[Function]}
        >
          <img
            alt="Curtains"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/1/4400.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Curtains
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S1E13
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/14"
          onClick={[Function]}
        >
          <img
            alt="Heads Will Roll"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10446.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Heads Will Roll
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E1
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/15"
          onClick={[Function]}
        >
          <img
            alt="Infestation"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10447.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Infestation
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E2
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/16"
          onClick={[Function]}
        >
          <img
            alt="Force Majeure"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10448.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Force Majeure
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E3
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/17"
          onClick={[Function]}
        >
          <img
            alt="Revelation"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10449.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Revelation
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E4
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/18"
          onClick={[Function]}
        >
          <img
            alt="Reconciliation"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10450.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Reconciliation
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E5
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/19"
          onClick={[Function]}
        >
          <img
            alt="In the Dark"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10451.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                In the Dark
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E6
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/20"
          onClick={[Function]}
        >
          <img
            alt="Going Home"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10452.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Going Home
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E7
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/21"
          onClick={[Function]}
        >
          <img
            alt="Awakening"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10453.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Awakening
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E8
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/22"
          onClick={[Function]}
        >
          <img
            alt="The Red Door"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10454.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                The Red Door
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E9
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/23"
          onClick={[Function]}
        >
          <img
            alt="The Fall"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10455.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                The Fall
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E10
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/24"
          onClick={[Function]}
        >
          <img
            alt="Black Ice"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10456.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Black Ice
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E11
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/25"
          onClick={[Function]}
        >
          <img
            alt="Turn"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10457.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Turn
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E12
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/26"
          onClick={[Function]}
        >
          <img
            alt="Go Now"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/4/10458.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Go Now
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S2E13
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/142270"
          onClick={[Function]}
        >
          <img
            alt="Move On"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/12/31233.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Move On
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E1
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/151048"
          onClick={[Function]}
        >
          <img
            alt="But I'm Not"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/12/31234.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                But I'm Not
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E2
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/151645"
          onClick={[Function]}
        >
          <img
            alt="Redux"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/12/31939.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Redux
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E3
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/153120"
          onClick={[Function]}
        >
          <img
            alt="The Kinship"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/12/32457.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                The Kinship
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E4
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/176889"
          onClick={[Function]}
        >
          <img
            alt="Alaska"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/13/33494.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Alaska
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E5
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/176890"
          onClick={[Function]}
        >
          <img
            alt="Caged"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/13/34493.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Caged
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E6
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/185048"
          onClick={[Function]}
        >
          <img
            alt="Ejecta"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/13/34948.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Ejecta
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E7
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/185049"
          onClick={[Function]}
        >
          <img
            alt="Breaking Point"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/14/36036.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Breaking Point
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E8
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/185050"
          onClick={[Function]}
        >
          <img
            alt="Plan B"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/14/37358.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Plan B
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E9
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/185051"
          onClick={[Function]}
        >
          <img
            alt="Legacy"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/15/39057.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Legacy
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E10
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/185052"
          onClick={[Function]}
        >
          <img
            alt="Love is a Battlefield"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/16/40250.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Love is a Battlefield
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E11
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/185053"
          onClick={[Function]}
        >
          <img
            alt="Incandescence"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/16/41932.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                Incandescence
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E12
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
    >
      <div
        className="MuiGridListTile-tile"
      >
        <a
          href="/episodes/185054"
          onClick={[Function]}
        >
          <img
            alt="The Enemy Within"
            src="http://static.tvmaze.com/uploads/images/medium_landscape/17/43622.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                The Enemy Within
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                S3E13
              </div>
            </div>
          </div>
        </a>
      </div>
    </li>
  </ul>
</div>
`;
